master/internal/rm/multirm/multirm_intg_test.go

```diff
@@ -632,6 +634,63 @@ func TestDisableSlot(t *testing.T) {
 	}
 }
 
+func TestVerifyNamespaceExists(t *testing.T) {
+	rm1 := mocks.ResourceManager{}
+	rm1ClusterName := "cluster1"
+	rm2ClusterName := "cluster2"
+	rm2 := mocks.ResourceManager{}
+	mockMultiRM := MultiRMRouter{
+		rms: map[string]rm.ResourceManager{
+			rm1ClusterName: &rm1,
+			rm2ClusterName: &rm2,
+		},
+		syslog: logrus.WithField("component", "resource-router"),
+	}
+
+	validNamespaceName := "good-namespace"
+	invalidNamespaceName := "bad-namespace"
+	cases := []struct {
+		name          string
+		namespaceName string
+		clusterName   string
+		setupMockRM   func()
+		err           error
+	}{
+		{
+			"valid-namespace",
+			validNamespaceName,
+			rm1ClusterName,
+			func() {
+				rm1.On("VerifyNamespaceExists", validNamespaceName, rm1ClusterName).
+					Return(nil).Once()
+			},
+			nil,
+		},
+		{
+			"invalid-namespace-name",
+			invalidNamespaceName,
+			rm2ClusterName,
+			func() {
+				rm2.On("VerifyNamespaceExists", invalidNamespaceName, rm2ClusterName).
+					Return(fmt.Errorf("namespace %s does not exist", invalidNamespaceName)).Once()
+			},
+			fmt.Errorf("namespace %s does not exist", invalidNamespaceName),
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			test.setupMockRM()
+			err := mockMultiRM.VerifyNamespaceExists(test.namespaceName, test.clusterName)
+			if err == nil {
+				require.Equal(t, test.err, err)
+			} else {
+				require.True(t, strings.Contains(err.Error(), "does not exist"))
```

Comment:
Comparing string contents of error messages is bad practice; is there a way to detect this case through something like `errors.Is`?